test/csuite/wt5212_incremental_backup/main.c

```diff
@@ -0,0 +1,779 @@
+         */
+        testutil_check(__wt_snprintf(buf, sizeof(buf), "rm -rf %s.%d && mkdir -p %s.%d/%s",
+          home_incr, i, home_incr, i, logpath));
+        testutil_check(system(buf));
+        if (i == 0)
+            continue;
+        /*
+         * For full backups we need 1-N.
+         */
+        testutil_check(__wt_snprintf(buf, sizeof(buf), "rm -rf %s.%d && mkdir -p %s.%d/%s",
+          home_full, i, home_full, i, logpath));
+        testutil_check(system(buf));
+    }
+}
+
+static void
+add_work(WT_SESSION *session, int iter, int iterj, const char *t_uri1, const char *t_uri2)
+{
+    WT_CURSOR *cursor, *cursor2;
+    int i;
+    char k[64], v[64];
+
+    testutil_check(session->open_cursor(session, t_uri1, NULL, NULL, &cursor));
+
+    /*
+     * Only on even iterations add content to the extra table. This illustrates and shows that
+     * sometimes only some tables will be updated.
+     */
+    cursor2 = NULL;
+    if (iter % 2 == 0) {
+        testutil_check(session->open_cursor(session, t_uri2, NULL, NULL, &cursor2));
+    }
+    /*
+     * Perform some operations with individual auto-commit transactions.
+     */
+    for (i = 0; i < MAX_KEYS; i++) {
+        testutil_check(__wt_snprintf(k, sizeof(k), "key.%d.%d.%d", iter, iterj, i));
+        testutil_check(__wt_snprintf(v, sizeof(v), "value.%d.%d.%d", iter, iterj, i));
+        cursor->set_key(cursor, k);
+        cursor->set_value(cursor, v);
+        testutil_check(cursor->insert(cursor));
+        if (cursor2 != NULL) {
+            cursor2->set_key(cursor2, k);
+            cursor2->set_value(cursor2, v);
+            testutil_check(cursor2->insert(cursor2));
+        }
+    }
+    testutil_check(cursor->close(cursor));
+    if (cursor2 != NULL)
+        testutil_check(cursor2->close(cursor2));
+}
+
+static void
+add_bulk_load(WT_SESSION *session, int iter_i, int iter_j, const char *t_uri)
+{
+    WT_CURSOR *cursor;
+    int i;
+    int k;
+    char v[64];
+
+    testutil_check(session->open_cursor(session, t_uri, NULL, "bulk", &cursor));
+
+    for (i = 0; i < MAX_KEYS; i++) {
+        k = iter_i + iter_j + i;
+        testutil_check(__wt_snprintf(v, sizeof(v), "value.%d.%d.%d", iter_i, iter_j, i));
+
+        cursor->set_key(cursor, k);
+        cursor->set_value(cursor, v);
+        testutil_check(cursor->insert(cursor));
+    }
+    testutil_check(cursor->close(cursor));
+}
+
+static void
+remove_work(WT_SESSION *session, int iter_i, int iter_j)
+{
+    WT_CURSOR *cursor;
+    int i;
+    char k[64], v[64];
+
+    testutil_check(session->open_cursor(session, uri, NULL, NULL, &cursor));
+
+    /*
+     * Remove records from the main table.
+     */
+    for (i = 0; i < MAX_KEYS; i++) {
+        testutil_check(__wt_snprintf(k, sizeof(k), "key.%d.%d.%d", iter_i, iter_j, i));
+        testutil_check(__wt_snprintf(v, sizeof(v), "value.%d.%d.%d", iter_i, iter_j, i));
+        cursor->set_key(cursor, k);
+        cursor->set_value(cursor, v);
```

Comment:
The `remove` call only needs a key, not a value.

Reply:
Fixed